### ISSUE TITLE
Plugin does nothing when clicked

### ISSUE THREAD

**Issue by sjd753** [appdictive/ADBRemoteConnect#5]
Plugin does nothing when clicked. 

Note: There is no Oracle java installed. New Android Studio comes pre-loaded with Open JDK. Hence there is no need to set environment variable for adb. Could it be the reason?

**Comment by JesperLR** [comment]
Yes that sounds likely - do you not get an Android Studio warning notification that ADB is missing?
ADBWiFiConnect should spawn such a notification dialog within Android Studio if ADB path is not found by the Java AndroidSDKUtils that it uses.

**Comment by sjd753** [comment]
Nope! No warnings in Android Studio 2.3.3

**Comment by snj33v** [comment]
nothing happens here too, no error too

**Comment by sytolk** [comment]
The same issue for me too. I'm using Intellij Idea v. 2017.2.5 on Debian host.
Works for me with Windows and Android Studio 2.3.3

**Comment by JesperLR** [comment]
Think I'll need some more log info - can I get someone to try this version on a OS where it does not launch? (in Android Studio go to Settings -> Plugins -> Install plugin from disk... - then select the attached zip with version 1.7.1 and restart Studio
[ADBWiFiConnect-1.7.1.zip](https://github.com/appdictive/ADBWiFiConnect/files/1413689/ADBWiFiConnect-1.7.1.zip)
)
It should now make a Notification within IntelliJ/Android Studio stating the command it attempted to run the ADB WiFi Connect jar. My suspicion is that it might be the OS name that is different than expected - but this should give us the necessary info to figure it out :)

**Comment by sjd753** [comment]
Okay I'm gonna try this and get back here in a couple of days hopefully.

**Comment by sjd753** [comment]
Hey just got the logs.. here it is..

**Could not launch the ADB WiFi Connect jar. Please report this to the developer along with the attempted launch command: (windows 7) cmd.exe /c cd / && javaw -jar "C:\Users\AndroidDev\.AndroidStudio2.3\config\plugins\ADBWiFiConnect\lib\ADBWiFiConnectGUI.jar"**

I suppose same woulld happen in windows 10. 

I think I noticed that if Oracle Java is installed it works fine but if new Android Studio is downloaded it comes with OpenJDK and does not require Oracle java then it does not work!

**Comment by JesperLR** [comment]
Interesting - the command looks fine to me. What happens then if you attempt to run the command manually through commandprompt: 
cmd.exe /c cd / && javaw -jar "C:\Users\AndroidDev.AndroidStudio2.3\config\plugins\ADBWiFiConnect\lib\ADBWiFiConnectGUI.jar"

Or if you navigate to "C:\Users\AndroidDev.AndroidStudio2.3\config\plugins\ADBWiFiConnect\lib" do you then see the jar file?

Oh, it might be that you do not have 'javaw' available as a command at all then?

**Comment by sytolk** [comment]
From Linux OS and Intellij Idea:
`Could not launch the ADB WiFi Connect jar. Please report this to the developer along with the attempted launch command: (linux) /bin/sh -c cd / && java -jar "/home/stanimir/.IntelliJIdea2017.2/config/plugins/ADBWiFiConnect/lib/ADBWiFiConnectGUI.jar"`

$ javaw                                                                            
bash: javaw: command not found

$ java -version                                                                                                                                                               
openjdk version "1.8.0_141"                                                                                                                                                                              
OpenJDK Runtime Environment (build 1.8.0_141-8u141-b15-1~deb9u1-b15)                                                                                                                                     
OpenJDK 64-Bit Server VM (build 25.141-b15, mixed mode)

**Comment by JesperLR** [comment]
And what happens if you @sytolk run the entire command manually in bash?

/bin/sh -c cd / && java -jar "/home/stanimir/.IntelliJIdea2017.2/config/plugins/ADBWiFiConnect/lib/ADBWiFiConnectGUI.jar"

**Comment by sytolk** [comment]
$ /bin/sh -c cd / && java -jar "/home/stanimir/.IntelliJIdea2017.2/config/plugins/ADBWiFiConnect/lib/ADBWiFiConnectGUI.jar"
Error: Could not find or load main class dk.appdictive.adbwificonnect.Main

**Comment by JesperLR** [comment]
@sytolk oh, that's interesting, I'm not quite sure what's causing the issue on your system then.. what if you navigate to /home/stanimir/.IntelliJIdea2017.2/config/plugins/ADBWiFiConnect/lib are you able to run the .jar in bash at all? Do you have experience with running jar files from bash? :)

**Comment by sytolk** [comment]
$ java -jar ADBWiFiConnectGUI.jar 
Error: Could not find or load main class dk.appdictive.adbwificonnect.Main

Yes I have experience with running .jar :) but its appears that this jar is not packaged correctly ;)

**Comment by JesperLR** [comment]
@sytolk cool :) haha yeah something's differently not right - weird thing is that it runs fine on many other setups, so it seems your java environment expects something else from the jar.. I'll have a look at it asap, but might be a couple of days before I get a chance to dive into it

**Comment by sytolk** [comment]
@JesperLR ok :) If you cannot repeat this issue just ping me. I can fork this plugin repo and try to run project direct from IDE

**Comment by sjd753** [comment]
> Or if you navigate to "C:\Users\AndroidDev.AndroidStudio2.3\config\plugins\ADBWiFiConnect\lib" do you then see the jar file?

Yes I do see it there!

**Comment by JesperLR** [comment]
Alright, I think I've figured out the root of the problem - it's pretty much what it says here:
https://stackoverflow.com/questions/40267122/openjdk-cannot-find-main-class-in-jar-while-oraclejdk-can

OpenJDK does not support JavaFX, which is the Java code I am using to build the UI for the plugin - it comes bundled with the regular OracleJDK.
I am not entirely sure if there's much to do about it other than for me to try to identify if this is the issue when trying to run the plugin and give an appropriate error message encouraging the user to install OracleJDK (a.k.a. the standard Java installation) if they want to make the plugin run. 
Any input on this? :)

**Comment by kstenerud** [comment]
It would probably be better to just use normal Java for the UI. Right now the out-of-box experience is terrible:

- Install fresh Android Studio
- Go to plugins repository
- Install WIFI connect & restart
- Click button
- Nothing happens
- Uninstall plugin and give 1 star review "Doesn't work"

Popping up a request to install a different SDK is still major adoption friction that will drive users elsewhere. What makes or breaks a plugin is the "I click install and it just works" factor. Nobody wants to get stuck yak shaving.

**Comment by berkayturanci** [comment]
Plugin doesn't work in my second macOS. I could be able to run it on my first Mac computer however. I do not understand what is the differences. Anyway, just download the .zip file and start it from there. It works fine. 

> Think I'll need some more log info - can I get someone to try this version on a OS where it does not launch? (in Android Studio go to Settings -> Plugins -> Install plugin from disk... - then select the attached zip with version 1.7.1 and restart Studio
> [ADBWiFiConnect-1.7.1.zip](https://github.com/appdictive/ADBWiFiConnect/files/1413689/ADBWiFiConnect-1.7.1.zip)
> )
> It should now make a Notification within IntelliJ/Android Studio stating the command it attempted to run the ADB WiFi Connect jar. My suspicion is that it might be the OS name that is different than expected - but this should give us the necessary info to figure it out :)

**Comment by ejgraton** [comment]
JavaFX can be "injected" into Android Studio openJDK (Programs Files\Android...Studio\jre) by the following steps from here: https://github.com/SkyLandTW/OpenJFX-binary-windows/releases/tag/v8u172-b11

Additional step 1: create Prefs key using regedit (try first or second indicated paths): https://github.com/julienvollering/MIAmaxent/issues/1#issuecomment-278985033

Additional step 2: be sure that jre\bin folder is included in the Path !
